state/app/BUILD

```diff
@@ -31,6 +31,7 @@ kt_jvm_library(
     srcs = glob(["*.kt"]),
     deps = [
         "//state/common:common",
+        "//framework/common",
```

Comment:
Does this make sense? 

Framework depending on state is common throughout our code, but having our state in turn depend on framework common seems like an inversion of that and approaching a cycle. Obviously, this doesn't fail to build but that doesn't clear the bar for something being acceptable.

The logic for requiring this seems sound to me:
- We have a preference manager within `//state/app`, where we store all of our managers for application state.
- Within the preference manager we want to refer to properties of the program, including (in this case) the type `GraphTheme`.
- So we need to depend on `//framework/common`, where we store common aspects of our framework including our themes.

I've attempted to generate incorrect-looking dependency graphs using the tools in `/docs` but wasn't able to find anything that proved this.

Reply:
If we're in the situation where `state` is depending on `framework`, we have a problem. It is indeed a circular dependency and it means something is placed wrongly.

The most likely discrepancy is this: `Theme` _feels_ like part of the framework, because it's a purely visual thing. But perhaps the more rational way to see it is that it's a global setting - part of the global state - that has an impact on the appearance of the application.

In this way we can justify migrating `Theme` from `framework/common` over to somewhere in `state` - perhaps a new package `state/theme`.

Reply:
Hmm, that's definitely a reasonable approach. However, it seems as though the preference manager's expansion will end up justifying a lot of similar moves in the future. Let's discuss this in person when you're next available.

Reply:
After an internal discussion, we've decided we'll break this cycle by having preference manager store two sets of enums for higher level concepts on both the preference and the module/framework side. This way, when we want to perform translations in either direction, we still leverage the compiler to check that we have been exhaustive when we start adding types of, say, themes on either side.